include/nbl/builtin/hlsl/cpp_compat/matrix.h

```diff
@@ -0,0 +1,51 @@
+#ifndef _NBL_BUILTIN_HLSL_MATRIX_INCLUDED_
+#define _NBL_BUILTIN_HLSL_MATRIX_INCLUDED_
+
+#ifdef __cplusplus
+#include <glm/glm.hpp>
+#include <stdint.h>
+#endif
+
+namespace nbl::hlsl
+{
+
+#ifdef __cplusplus
+template<typename T, uint16_t N, uint16_t M>
+struct matrix final : private glm::mat<N,M,T>
+{
+    using Base = glm::mat<N,M,T>;
+    using Base::Base;
+    using Base::operator=;
+
+    template<uint16_t X, uint16_t Y> matrix(matrix<T, X, Y> const&) = delete;
```

Comment:
explain?

Reply:
I wasnt sure how or if hlsl handled matrix dimension conversions. When I figure out the correct way hlsl does the conversion I'll implement this accordingly.

Reply:
`enable_if` is your friend here if you need to check something about dimensions being less or more